app/models/deploy_group.rb

```diff
@@ -0,0 +1,9 @@
+class DeployGroup < ActiveRecord::Base
+  has_soft_deletion default_scope: true
+
+  belongs_to :environment
+  has_and_belongs_to_many :stages
+
+  validates_presence_of :name, :environment_id
```

Comment:
it's a speed (no extra load) vs correctness tradeoff, either way should be
fine though ...

On Tue, Mar 3, 2015 at 3:16 PM, Roman Sandler <notifications@github.com>
wrote:

> In app/models/deploy_group.rb
> <https://github.com/zendesk/samson/pull/251#discussion_r25737968>:
>
> > @@ -0,0 +1,9 @@
> > +class DeployGroup < ActiveRecord::Base
> > +  has_soft_deletion default_scope: true
> > +
> > +  belongs_to :environment
> > +  has_and_belongs_to_many :stages
> > +
> > +  validates_presence_of :name, :environment_id
>
> could validate the presence of the environment rather than environment_id:
> http://guides.rubyonrails.org/active_record_validations.html#presence
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/251/files#r25737968>.
>